### PR TITLE
Sequence diagram assert

### LINKED ISSUE #572: Support assert in sequence diagram descriptions
Assertions would make the creation of sequence diagram descriptions more convenient. As the team discussed at the last meeting, the goal is to let the users define assertions to the actual state of a statemachine. 

The syntax could be something like `Sequence.assertState(modelClassInstance, ConcreteState.class);`.

### PR DESCRIPTION
**Pull request by kmozsi**
This pull request resolves #572.

* State assertion in sequence diagram descriptions is supported in case of known senders and targets.
* Error hierarchy is remade to handle introduced warnings also.
* Removed unnecessary methods from `Sequence` class.

### CODE REVIEW

**Comment by kmozsi** [comment]
@djnemeth, @jtamas96, @nagyandras95, please review this pull request.

File: dev/plugins/hu.elte.txtuml.api.model.execution/src/hu/elte/txtuml/api/model/execution/impl/seqdiag/SeqDiagModelExecutorThread.java [hunk 1]
```diff
@@ -80,7 +82,21 @@ private void compareToPattern(SignalWrapper wrapper, ModelClass target, boolean
...
+	 * Verifies that the {@code ModelClass} instance is currently in the given
+	 * state. If the assertion fails, a {@link ModelStateAssertError} will be
+	 * added.
+	 */
+	public void assertState(ModelClass instance, Class<?> state) {
+		Class<?> currentState = ((SeqDiagModelClassRuntime) ((DefaultSeqDiagRuntime) getModelRuntime())
+				.getRuntimeOf(instance)).getCurrentState();
+		if (!currentState.equals(state)) {
+			root.getExecutor().addError(
+					new ModelStateAssertError(state.getCanonicalName(), currentState.getClass().getCanonicalName()));
```
[the remark this answers is not in the document]
Comment (kmozsi) [reply dev/plugins/hu.elte.txtuml.api.model.execution/src/hu/elte/txtuml/api/model/execution/impl/seqdiag/SeqDiagModelExecutorThread.java]: You are right, thanks!

File: dev/plugins/hu.elte.txtuml.api.model.execution/src/hu/elte/txtuml/api/model/execution/impl/seqdiag/SeqDiagModelExecutorThread.java [hunk 2]
```diff
@@ -80,7 +82,21 @@ private void compareToPattern(SignalWrapper wrapper, ModelClass target, boolean
...
-}
\ No newline at end of file
+
+	/**
+	 * Verifies that the {@code ModelClass} instance is currently in the given
+	 * state. If the assertion fails, a {@link ModelStateAssertError} will be
+	 * added.
+	 */
+	public void assertState(ModelClass instance, Class<?> state) {
+		Class<?> currentState = ((SeqDiagModelClassRuntime) ((DefaultSeqDiagRuntime) getModelRuntime())
```
Comment (djnemeth) [inline dev/plugins/hu.elte.txtuml.api.model.execution/src/hu/elte/txtuml/api/model/execution/impl/seqdiag/SeqDiagModelExecutorThread.java]: I suggest to break up these lines to named subexpressions by introducing temporary variables. Also, are these casts really necessary? That is, couldn't they be abstracted away with appropriate interface changes?

**Comment by djnemeth** [comment]
Jenkins, retest this, please.

**Comment by djnemeth** [comment]
Jenkins, retest this, please.

**Comment by djnemeth** [comment]
Jenkins, retest this, please.

**Pull request merged**